extension-files/bringweb3-sdk/utils/background/activate.ts

```diff
@@ -2,15 +2,25 @@ import storage from "../storage";
 import addQuietDomain from "./addQuietDomain";
 import checkNotifications from "./checkNotifications";
 import getCashbackUrl from "./getCashbackUrl";
+import isWhitelisted from "./isWhitelisted";
 
-const handleActivate = async (domain: string, extensionId: string, identifier: string, cashbackPagePath: string | undefined, time?: number) => {
+const handleActivate = async (domain: string, extensionId: string, identifier: string, cashbackPagePath: string | undefined, time?: number, tabId?: number, redirectUrl?: string) => {
     if (extensionId === chrome.runtime.id) {
         await storage.set('lastActivation', Date.now());
     }
 
     if (domain) addQuietDomain(domain, time || Date.now() + 24 * 60 * 60 * 1000)
 
+    if (tabId && redirectUrl) {
+        const whitelist = await storage.get('redirectsWhitelist')
+
+        if (!whitelist?.length || await isWhitelisted(redirectUrl, whitelist)) {
```

Comment:
Check for empty whitelist must be removed on our fork. Empty whitelist means a missing whitelist means no redirected can happen.